### PR TITLE
docs: Update DG future enchancement

### PR DESCRIPTION
**Pull request by KengHian**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -458,3 +458,25 @@ testers are expected to do more *exploratory* testing.
...
+  * Support other job duration attribute (6-month contract, summer vacation internship)
+  * Support job requirement type (mandatory, good to have)
+  * Support job quota
+  * Smart select company if not found when creating job listing (find & return 1st match)
+  * Support screening candidates via more advanced requirement (salary range, number of skills)
+  * Sorting capability of job list
+* Company
+  * Edit a company
+  * Locating company by name
+  * Enhance company ali
```
Comment (Green-Tea-123) [inline docs/DeveloperGuide.md:480]: What does ali mean?

**Review by Green-Tea-123 — APPROVED** [review]
LGTM.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (KengHian) [reply docs/DeveloperGuide.md:480]: Opps, it was supposed to be about company liaise, but I am not sure if I should include it.

Comment (Green-Tea-123) [reply docs/DeveloperGuide.md:480]: oh then you can just include it then. Basically just 'Enhance company liaising with the addition of hiring manager contacts' or something like that.

**Review by wassilililily — APPROVED** [review]
LGTM

**Pull request merged**